src/lang/KclSingleton.ts

```diff
@@ -406,13 +406,39 @@ export class KclManager {
   }
   async executeCode(zoomToFit?: boolean): Promise<void> {
     const ast = this.safeParse(codeManager.code)
+
     if (!ast) {
       this.clearAst()
       return
     }
+
+    zoomToFit = this.tryToZoomToFitOnCodeUpdate(ast, zoomToFit)
+
     this.ast = { ...ast }
     return this.executeAst({ zoomToFit })
   }
+  /**
+   * This will override the zoom to fit to zoom into the model if the previous AST was empty.
+   * Workflows this improves,
+   *  When someone comments the entire file then uncomments the entire file it zooms to the model
+   *  When someone CRTL+A and deletes the code then adds the code back it zooms to the model
+   *  When someone CRTL+A and copies new code into the editor it zooms to the model
+   */
+  tryToZoomToFitOnCodeUpdate(ast: Program, zoomToFit: boolean | undefined) {
```

Comment:
Below is a real tsc problem. Try this. You'll want to do this in your editor in case it has knock-on effects.

```suggestion
  tryToZoomToFitOnCodeUpdate(ast: Node<Program>, zoomToFit: boolean | undefined) {
```